blueocean-core-js/src/js/UrlBuilder.js

```diff
@@ -103,6 +103,11 @@ export const buildRunDetailsUrlFromQueue = (queueItem, isMultiBranch, expectedBu
     // if multi-branch, change the last value of 'pipelines' to 'branches' so it looks like a multibranch REST URL
     if (isMultiBranch) {
         tokens[tokens.length - 4] = 'branches';
+
+        // TODO: compensation until JENKINS-37746 is fixed - encode the branch name since the server only encoded it once
+        if (encodeBranchName) {
+            tokens[tokens.length - 3] = encodeURIComponent(tokens[tokens.length - 3]);
+        }
```

Comment:
I don't see any `tokens.length` checks before accessing. Guaranteed to not be needed?